pkg/puller/downloader.go

```diff
@@ -0,0 +1,172 @@
+package puller
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"hash/fnv"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var (
+	d *Downloader
+)
+
+type Downloader struct {
+	NumRetries int
```

Comment:
I think we probably never want to stop retrying. Keep retrying, but backoff to some slow interval if it keeps failing. These systems need to self heal.